### PR TITLE
Support global character initialization and initialization with logical and equality operators

### PR DESCRIPTION
**Pull request by AW-AlanWu**
This PR fixes the issue where global assignments cannot assign characters and do not support logical and equality operators. Corresponding tests have been added to ensure these changes.

### CODE REVIEW

**Review by jserv — CHANGES_REQUESTED** [review]
Check https://cbea.ms/git-commit/ carefully and refine.

**Review by ChAoSUnItY — APPROVED** [review]

**Review by vacantron — APPROVED** [review]

File: src/parser.c [hunk 1]
```diff
@@ -1879,6 +1885,18 @@ int eval_expression_imm(opcode_t op, int op1, int op2)
...
         res = op1 >> op2;
         break;
+    case OP_log_and:
+        res = op1 && op2;
+        break;
+    case OP_log_or:
+        res = op1 || op2;
+        break;
+    case OP_eq:
+        res = op1 == op2 ? 1 : 0;
```
Comment (DrXiao) [inline src/parser.c:1895]: If possible, directly rewrite it as `res = op1 == op2` to avoid generating branch instructions.

File: src/parser.c [hunk 2]
```diff
@@ -1879,6 +1885,18 @@ int eval_expression_imm(opcode_t op, int op1, int op2)
...
+        break;
+    case OP_neq:
+        res = op1 != op2 ? 1 : 0;
```
Comment (DrXiao) [inline src/parser.c:1898]: Ditto.

**Comment by jserv** [comment]
Thank @AW-AlanWu for contributing!

**Pull request merged**